docs/DeveloperGuide.md

```diff
@@ -486,367 +498,250 @@ Hence, any modification to an employee after SudoHR is initialized from storage
+| `* * *`  | HR personnel | edit a department                                              | ensure consolidation of information when a department's detail is changed  |
+| `* * *`  | HR personnel | delete a department                                            | ensure consolidation of information when a department is disbanded         |
+| `* * *`  | HR personnel | find a department by name                                      | locate details of departments without having to go through the entire list |
+| `* * *`  | HR personnel | add an employee to a department                                | ensure consolidation of information when a department has a new employee   |
+| `* * *`  | HR personnel | remove an employee from a department                           | ensure consolidation of information when an employee leaves a department   |
+| `* * *`  | HR personnel | list all departments an employee is in                         |                                                                            |
+| `* * *`  | HR personnel | list all employees in a department                             | view manpower size of a department                                         |
+
+_{More to be added}_
 
 ### 6.3. Use cases
 
-(For all use cases below, the **System** is  `SudoHR` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is `SudoHR` and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: Add a new employee**
+**Use case: Listing all departments**
 
 **MSS**
 
-1.  User requests to list employees
-2.  SudoHR shows a list of employees
-3.  User requests to delete a specific employee in the list
-4.  SudoHR deletes the employee
+1. User requests to list all departments
+2. SudoHr shows the list of all departments
 
-    Use case ends.
+   Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
-
-  Use case ends.
-
-* 3a. The given index is invalid.
-
-    * 3a1. SudoHR shows an error message.
+- 1a. The list of departments is empty.
 
-      Use case resumes at step 2.
+  use case ends
 
-**Use case: UC1 - Create a department**
+**Use case: Finding a department using a keyword**
 
-**MSS:**
-1. User requests to create a department.
-2. SudoHR creates the department.
+1. User requests to show the list of departments containing specified keywords
+2. SudoHr shows the list of all departments with name that contains any of the keywords.
 
-    Use case ends.
-
-**Extensions**
-
-* 1a. The given argument is invalid.
-
-    * 1a1. SudoHR shows an error message.
-
-      Use case resumes at step 1.
-
-**Use case: UC2 - Add employee to department**
-**MSS:**
-1. User lists department
-2. User requests to add employee to the department.
-3. SudoHR adds employee to the department.
-4. Use case ends.
+   Use case ends.
 
 **Extensions**
 
-* 1a. The given argument is invalid.
+- 1a. There are no departments with any of the keywords.
```

Comment:
Assuming if the list of departments shows an empty list if no department has the keyword, then this extension should not exist since step 2 will still occur just that empty list will be shown

Reply:
Yep you are right, will fix this